inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/initializers/FactLayerInitializer.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2018
+ * Ubiquitous Knowledge Processing (UKP) Lab and FG Language Technology
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.inception.ui.kb.initializers;
+
+import static de.tudarmstadt.ukp.clarin.webanno.api.WebAnnoConst.SPAN_TYPE;
+import static java.util.Arrays.asList;
+
+import java.io.IOException;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.AnnotationSchemaService;
+import de.tudarmstadt.ukp.clarin.webanno.api.dao.initializers.LayerInitializer;
+import de.tudarmstadt.ukp.clarin.webanno.api.dao.initializers.ProjectInitializer;
+import de.tudarmstadt.ukp.clarin.webanno.api.dao.initializers.TokenLayerInitializer;
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationFeature;
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationLayer;
+import de.tudarmstadt.ukp.clarin.webanno.model.LinkMode;
+import de.tudarmstadt.ukp.clarin.webanno.model.MultiValueMode;
+import de.tudarmstadt.ukp.clarin.webanno.model.Project;
+import de.tudarmstadt.ukp.dkpro.core.api.ner.type.NamedEntity;
+import de.tudarmstadt.ukp.inception.ui.kb.feature.FactLinkingConstants;
+import de.tudarmstadt.ukp.inception.ui.kb.feature.PropertyFeatureSupport;
+
+@Component
+public class FactLayerInitializer
+    implements LayerInitializer
+{
+    private final AnnotationSchemaService annotationSchemaService;
+
+    @Autowired
+    public FactLayerInitializer(AnnotationSchemaService aAnnotationSchemaService)
+    {
+        annotationSchemaService = aAnnotationSchemaService;
+    }
+
+    @Override
+    public List<Class<? extends ProjectInitializer>> getDependencies()
+    {
+        // Because locks to token boundaries
+        return asList(TokenLayerInitializer.class);
+    }
+
+    @Override
+    public void configure(Project aProject)
+        throws IOException
+    {
+        AnnotationLayer factLayer = new AnnotationLayer(FactLinkingConstants.FACT_LAYER, "Fact",
+            SPAN_TYPE, aProject, false);
+        factLayer.setAllowStacking(true);
+        factLayer.setCrossSentence(false);
+        factLayer.setLockToTokenOffset(false);
+        factLayer.setMultipleTokens(true);
+
+        annotationSchemaService.createFeature(
+            new AnnotationFeature(aProject, factLayer, "KBPredicate", "a-Predicate",
```

Comment:
Feature names should start with a small letter.

Feature names: `predicate`, `subject`, `object', `qualifiers`.

If you really need to prefix the UI names, I would suggest to prefix them with numbers, e.g. `1) Predicate`.

Reply:
I cannot use numbers as prefix because "Feature names must start with a letter and consist only of letters, digits, or underscores".

Reply:
That limitation exists if you create a feature in the UI and it only exists when you initially created the feature because at that point, the "internal name" is derived from the feature name. After the feature has been created, you can rename it arbitrarily, because then it only affects the "uiName". When you use the API directly, you set the "name" and "uiName" independently anyway, so you are not bound by this limitation. The "name" should start with a lower-case letter and must conform to the Java naming rules for class fields. The "uiName" can essentially be anything.

Reply:
Thanks, I have changed them. BTW, if I rename the name manually in projects setting, I still get this message and cannot rename it arbitrarily.